faiss/python/loader.py

```diff
@@ -3,30 +3,48 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
+from distutils.version import LooseVersion
 import platform
 import subprocess
 import logging
 
 
 def instruction_set():
+    """
+    Returns a dictionary for supported instruction sets, see
+    https://github.com/numpy/numpy/blob/master/numpy/core/src/common/npy_cpu_features.h
+    for the list of features that this dictionary contains per architecture.
+
+    Example:
+    >>> instruction_set()  # for x86
+    {"SSE2": True, "AVX2": False, ...}
+    >>> instruction_set()  # for PPC
+    {"VSX": True, "VSX2": False, ...}
+    >>> instruction_set()  # for ARM
+    {"NEON": True, "ASIMD": False, ...}
+    """
```

Comment:
I know this is constrained on the numpy side but what about returning a set with the flags set to True? 

For example, below the has_AVX2 check will fail on ARM with `KeyError`.

Reply:
Good point about the key error. I think the cleanest fix would just be to use `.get("AVX2", False)` on the output, which is fine even if some keys are not there.

Alternatively, we could "pad" the dictionary with the keys from the other architectures with value `False`. I don't like this idea though, because we'd be duplicating the list of instruction sets compared to numpy

Reply:
We could also return a set in every case (in the case of numpy, selecting the keys for which the value is `True`).

Reply:
Sure, that's a good idea as well.

Reply:
Thanks for using the set. Will get it through the FB internal checks.